From 040fafaf7260b51031e20d850aa5358c0a7bd9a7 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Sat, 5 Nov 2022 22:30:07 +0100
Subject: [PATCH] build: unbreak compilation on Linux
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gzdoom/src/d_anonstats.cpp:14:10: fatal error: i_mainwindow.h: No such file or directory
   14 | #include "i_mainwindow.h"
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/zdoom.dir/build.make:1441: src/CMakeFiles/zdoom.dir/d_anonstats.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:959: src/CMakeFiles/zdoom.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

gzdoom/src/d_anonstats.cpp:126:17: error: ‘actor’ was not declared in this scope; did you mean ‘AActor’?
  126 |                 actor->         return false; // no host, disable
      |                 ^~~~~
      |                 AActor

$ find . -name "i_mainw*"
./src/common/platform/win32/i_mainwindow.h
./src/common/platform/win32/i_mainwindow.cpp

Fixes: g4.5pre-1327-g6670bc09b
Fixes: g4.9pre-359-g082818405
---
 src/d_anonstats.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/d_anonstats.cpp b/src/d_anonstats.cpp
index 1dcd05824..2e1456259 100644
--- a/src/d_anonstats.cpp
+++ b/src/d_anonstats.cpp
@@ -11,9 +11,8 @@ void D_ConfirmSendStats()
 
 #else // !NO_SEND_STATS
 
-#include "i_mainwindow.h"
-
 #if defined(_WIN32)
+#include "i_mainwindow.h"
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 #include <winsock2.h>
@@ -121,7 +120,7 @@ bool I_HTTPRequest(const char* request)
 bool I_HTTPRequest(const char* request)
 {
 	if ((*sys_statshost)[0] == 0)
-		actor->		return false; // no host, disable
+		return false; // no host, disable
 
 	int sockfd, portno, n;
 		struct sockaddr_in serv_addr;
-- 
2.38.0

